packages/hash/frontend/src/blocks/page/BlockHighlightContext.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+  createContext,
+  FunctionComponent,
+  ReactNode,
+  useContext,
+  useMemo,
+  useState,
+} from "react";
+
+/** @private enforces use of custom provider */
+const BlockHighlightContext = createContext<{
+  highlightedBlockId: string;
+  setHighlightedBlockId: (blockId: string) => void;
+} | null>(null);
+
+type BlockHighlightProviderProps = {
+  children?: ReactNode;
+  routeHash: string;
+};
+
+export const BlockHighlightProvider: FunctionComponent<
```

Comment:
Since this relates a bit with `BlockLoadedProvider` (which is responsible for scrolling the block into view on load), I think it's better we move this logic there. That way we don't need a new context provider

Reply:
Moved the logic into `BlockLoadedProvider` in https://github.com/hashintel/hash/pull/914/commits/5909d84bdf9b3c072f9880ead908021bf9f56106

Reply:
@yusufkinatas I thought you used a separate context because we were going to highlight blocks under more conditions than just block loading, e.g. when the context menu for a given block is open. In which case the 'highlighted' status isn't just about the block being loaded. 

I think the latest approach works for now though, and we can rethink it when we introduce more conditions for block highlighting.

Reply:
Yes, we are probably going to separate the logic when we implement the highlighting for more scenarios like (e.g internal links, indicator when context menu is opened for blocks). For now this looks logical to me as you said, since it's only used on page loads like you said.